helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -241,13 +244,87 @@ private void refreshClusterConfig(final HelixDataAccessor accessor,
+          .format("No ClusterConfig change for cluster %s, pipeline %s", _clusterName,
+              getPipelineName()));
     }
   }
 
+  // TODO: This function is used to clean up incompatible batched disabled instances format for
+  // "DISABLED_INSTANCES" introduced in 1.0.3.0. This temp change shoul dbe reverted after 1.0.5.0 \
+  // or later version.
+  private boolean updateBatchDisableFormat(final HelixDataAccessor accessor) {
+    return accessor
+        .updateProperty(accessor.keyBuilder().clusterConfig(), new DataUpdater<ZNRecord>() {
+          @Override
+          public ZNRecord update(ZNRecord currentData) {
+            if (currentData == null) {
+              throw new HelixException(
+                  "Cluster: " + _clusterConfig.getClusterName() + ": cluster config is null");
+            }
+
+            ClusterConfig clusterConfig = new ClusterConfig(currentData);
+            Map<String, String> disabledInstances = clusterConfig.getDisabledInstances();
+            Map<String, String> DisabledInstancesWithInfo =
+                clusterConfig.getDisabledInstancesWithInfo();
+
+            ClusterConfig newClusterConfig = new ClusterConfig(currentData);
+            Map<String, String> newDisabledInstances =
+                new TreeMap<>(newClusterConfig.getDisabledInstances());
+            Map<String, String> newDisabledInstancesWithInfo =
+                new TreeMap<>(newClusterConfig.getDisabledInstancesWithInfo());
+
+            // iterate through all k,v pairs and check for string format.
+            for (Map.Entry<String, String> instanceInfo : disabledInstances.entrySet()) {
+              String instanceName = instanceInfo.getKey();
+              if (!StringUtils.isNumeric(instanceInfo.getValue())) {
+                newDisabledInstances.put(instanceName,
+                    ConfigStringUtil.parseConcatenatedConfig(instanceInfo.getValue()).get(
+                        ClusterConfig.ClusterConfigProperty.HELIX_ENABLED_DISABLE_TIMESTAMP
+                            .toString()));
+                newDisabledInstancesWithInfo.put(instanceName, instanceInfo.getValue());
+              } else {
+                if (!DisabledInstancesWithInfo.containsKey(instanceName)) {
+                  newDisabledInstancesWithInfo.put(instanceName, ConfigStringUtil
+                      .concatenateMapping(Collections.singletonMap(
+                          ClusterConfig.ClusterConfigProperty.HELIX_ENABLED_DISABLE_TIMESTAMP
+                              .toString(), instanceInfo.getValue())));
+                }
+              }
+            }
+            newClusterConfig.setDisabledInstances(newDisabledInstances);
+            newClusterConfig.setDisabledInstancesWithInfo(newDisabledInstancesWithInfo);
+            return newClusterConfig.getRecord();
+          }
+        }, null);
+  }
+
+  private boolean checkBatchedDisabledInstanceFormat(ClusterConfig clusterConfig) {
```

Comment:
Can we have a better method name? (if we still need this method) "check" might be too generic and it's unclear what it really means.